plutus-use-cases/test/Spec.hs

```diff
@@ -0,0 +1,33 @@
+module Main (main) where
+
+
+import           Test.Tasty
+import           Test.Tasty.Golden
+
+import qualified Data.ByteString.Lazy                                as BSL
+import           Data.Text.Encoding                                  (encodeUtf8)
+
+import           Language.Plutus.Coordination.Contracts.CrowdFunding
+import           Language.Plutus.Coordination.Plutus
+import           Language.Plutus.CoreToPLC.Plugin                    (PlcCode, getAst)
+import           Language.PlutusCore                                 (debugText)
+
+main :: IO ()
+main = defaultMain tests
+
+golden :: String -> PlcCode -> TestTree
+golden name value = (goldenVsString name ("test/" ++ name ++ ".plc.golden") . pure . BSL.fromStrict . encodeUtf8 . debugText . getAst) value
+
+tests :: TestTree
+tests = testGroup "Validator scripts" [
+    golden "crowdfunding" crowdfunding
+  ]
+
+crowdfunding :: PlcCode
+crowdfunding = getPlutusTx $ contributionScript c PubKey where
```

Comment:
I think I said before - but I'm not sure it's actually worth checking in the script. It's too big to be even remotely hand-verifiable. If we can write a test that it *evalutates* to the right thing then I think that's useful, otherwise let's not bother.